cmake/toolchain_handling.cmake

```diff
@@ -0,0 +1,29 @@
+if (CMAKE_TOOLCHAIN_FILE AND EXISTS ${CMAKE_TOOLCHAIN_FILE})
+  message("==> Loading \${CMAKE_TOOLCHAIN_FILE} == ${CMAKE_TOOLCHAIN_FILE}")
+
+else()
+  
+  set (CMAKE_SYSTEM_NAME "Generic")
+  
+  if ("${CROSS_COMPILE}" STREQUAL "")
+    set (CROSS_COMPILE "arm-none-eabi-")
+  endif()
+
+  include(CMakeForceCompiler)
+  cmake_force_c_compiler(  "${CROSS_COMPILE}gcc" GNU)
```

Comment:
Took me some time to understand what you mean here.
I've run it with cmake version 3.5.0-rc3 without any problems and without any warning.

Then I found: the deprecation is only mentioned here:
https://cmake.org/cmake/help/v3.5/module/CMakeForceCompiler.html?highlight=deprecated

It's not deprecated in 3.5.x yet!!

So maby [this variant](https://github.com/arobenko/embxx_on_rpi/pull/5) is good after all. But one should edit `src/CMakeLists.txt` to be similar to the change in this pull-req here.

Or: leave it as is, and fix it once we find out what the CMake team has decided to do. (Possible using a cmake-version conditional...)